### ISSUE TITLE
Missing flags for invalid use of aria-hidden

### ISSUE THREAD

**Issue by scottaohara** [validator/validator#1115]
This [test case](https://w3c.github.io/html-aria/tests/aria-hidden-invalid-element-test.html) includes element where global aria attributes are not allowed, and thus should flag use of aria-hidden as invalid.

per our discussion in #1095, I understand if a decision was made not to flag such errors in the checker since many of these elements are not directly exposed to users.  However, I would submit that if a warning isn't flagged for all these elements, `html`, `title`, `picture`, and `map` would need to flag this error.

thanks

**Comment by sideshowbarker** [comment]
> per our discussion in #1095, I understand if a decision was made not to flag such errors in the checker since many of these elements are not directly exposed to users.

To be clear: There’s no criteria in place to skip reporting errors for elements not being directly exposed to users. And in fact for the test case in https://w3c.github.io/html-aria/tests/aria-hidden-invalid-element-test.html, the checker does actually report an error for the `slot` element.

But the only reason the checker reports anything about `slot` in that case is, it’s a broken/invalid use for `slot` to begin with — because in that test case, the `slot` isn’t a descendant of a `template` element (and therefore it has zero effect of any kind).

There’s no unbroken/valid use of `slot` in which it would actually get parsed into the document content (DOM) of a document — rather than into a document fragment (separate from the DOM).

In contrast, the reason the checker doesn’t report anything for the `slot` element in the https://w3c.github.io/html-aria/tests/slot.html test is, that one is is actually an unbroken/valid use of `slot` — which means that the `slot` element doesn’t end up in the document content of the document, and therefore the checker never sees it at all and can’t do any checking of it.
